src/order/monovary.lean

```diff
@@ -268,3 +308,80 @@ protected lemma antivary_on_comm : antivary_on f g s ↔ antivary_on g f s :=
 ⟨antivary_on.symm, antivary_on.symm⟩
 
 end linear_order
+
+section
+variables [linear_order α] [linear_order β] (f : ι → α) (g : ι → β) {s : set ι}
+
+/-- If `f : ι → α` and `g : ι → β` are monovarying, then `monovary_order f g` is a linear order on
+`ι` that makes `f` and `g` simultaneously monotone. -/
+def monovary_order (i j : ι) : Prop :=
+prod.lex (<) (prod.lex (<) well_ordering_rel) (f i, g i, i) (f j, g j, j)
+
+instance : is_strict_total_order ι (monovary_order f g) :=
+{ trichotomous := λ i j, begin
+    convert trichotomous_of (prod.lex (<) $ prod.lex (<) well_ordering_rel) _ _,
+    { simp only [prod.ext_iff, ←and_assoc, imp_and_distrib, eq_iff_iff, iff_and_self],
+      exact ⟨congr_arg _, congr_arg _⟩ },
+    { apply_instance }
+  end,
+  irrefl := λ i, by { rw monovary_order, exact irrefl _ },
+  trans := λ i j k, by { rw monovary_order, exact trans } }
+
+variables {f g}
+
+lemma monovary_on_iff_exists_monotone_on :
+  monovary_on f g s ↔ ∃ [linear_order ι], by exactI monotone_on f s ∧ monotone_on g s :=
```

Comment:
This feels like it might be a pretty annoying spelling if `ι` already has an order.

What are your thoughts on making a `with_mono_vary_order ι` type synonym and instead stating this as something like 
```suggestion
  monovary_on f g s ↔ monotone_on (f ∘ of_mvo)  (of_mvo ⁻¹' s) ∧ monotone_on (g ∘ of_mvo)  (of_mvo ⁻¹' s) :=
```

Reply:
I thought about it too, but I would need four different type synonyms for the lemmas here, so I'm not convinced. The use case is deducing the Chebyshev sum inequality (pretty general and stated in terms of `monovary` in #13187) from the FKG inequality, where everything has an order (and is thus a priori less general). So the only point at which I will have this extra `linear_order ι` instance will be locally in the proof of Chebyshev, where there is no pre-existing order on `ι`.

Reply:
Why do you need four type synonyms?

Reply:
because the same type won't work for all four lemmas (I would need to insert `order_dual` in places).

Reply:
I'd probably argue that at that point you don't need all four lemmas; only the two(?) that don't mention order_dual at all.

Reply:
I don't think that's an improvement over the current situation given my use case and the todo I am leaving (would you want yet another type synonym for families of functions?).

Reply:
In fact, I see it as such:
* If I provide a type synonym, I make the choice for the user on what order they want to use (regarding `order_dual`).
* If the user expands out my `∃ [linear_order ι], ...`, they are free to make it a local instance, or place it onto a type synonym (which you indeed can create locally using `set` and `clear_value`).